### ISSUE TITLE
What is expected effect of setting Request type and destination? For example, to "audio"?

### ISSUE THREAD

**Issue by guest271314** [whatwg/fetch#555]
What is expected effect of setting `Request` [`type`](https://fetch.spec.whatwg.org/#concept-request-type) and [`destination`](https://fetch.spec.whatwg.org/#concept-request-destination)? For example, to `"audio"` when requesting an audio resource? 

[5.3. Request class](https://fetch.spec.whatwg.org/#request-class)
```

  readonly attribute RequestType type;
  readonly attribute RequestDestination destination;

```

**Comment by wanderview** [comment]
Javascript code cannot set the `type` or `destination` properties.  They are readonly and the `RequestInit` dictionary does not support them.

These properties are set by the browser for internally generated requests in order to apply various policies to the request.  For example, some types/destinations are allowed mixed content while others are not, etc.

**Comment by guest271314** [comment]
@wanderview What do you mean by "set by the browser for internally generated requests"? The browser uses `Request` internally? 

The context is an attempt to use `fetch()` to request a media resource with media fragment uri concatenated to URL string, where expected result is for only that portion of the partial content to be returned at `.blob()` or `.arrayBuffer()` call. Probably not possible, though that is what tried, here, when read the specification where `type` and `destination` were mentioned. 

Related https://github.com/whatwg/fetch/issues/554

**Comment by wanderview** [comment]
> What do you mean by "set by the browser for internally generated requests"? The browser uses Request internally?

The browser uses an internal `request` that is reflected to javascript as the `Request` DOM object.  The type and destination are set directly on the internal request.  The rules about what to set and when are summarized in the table here:

https://fetch.spec.whatwg.org/#concept-request-destination

**Comment by guest271314** [comment]
@wanderview  Yes. Does `HTMLMediaElement` call internal `request` at 

`<audio controls src="https://upload.wikimedia.org/wikipedia/commons/b/be/Hidden_Tribe_-_Didgeridoo_1_Live.ogg#t=30,50"></audio>
`
?

How can we port the media fragment uri functionality to `Request`?

**Comment by wanderview** [comment]
It should, but we don't implement `type` or `destination` yet in gecko.  I'm not sure about chrome.  This part of the spec went through some changes so implementation has lagged.  In the past it was called `context` and there was a lot of confusion about what values it should contain.

I'm not sure I know enough about media fragments to say, but it seems @jakearchibald has suggested we should just better support range requests.

**Comment by guest271314** [comment]
@wanderview The table was confusing to this reader at specification. It is there; try to set those values when have a case where trying to set a media fragment uri. Not sure if a proxy could be set, that is, make request from `Request`, `fetch()` be equivalent to `new Audio()`. Was merely curious about references to `type` and `destination` at specification, what they do and what the context and application of the properties.

**Comment by annevk** [comment]
Closing as this doesn't appear to be a problem with Fetch.

**Comment by annevk** [comment]
(Except for type/destination confusion, which is being addressed in #581.)
